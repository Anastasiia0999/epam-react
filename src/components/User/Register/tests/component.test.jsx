import React from 'react';
import { mount } from 'enzyme';

import {Register} from "../index";

describe('<Register />', () =>{
    let wrapper;
    let instance;

    const props = {
        fields:{
            "alert-success": "Successful pass",
            "alert-error": "Invalid data",
            "logIn-name": "Login",
            "password-name": "Password",
            "logIn-placeholder": "Input login",
            "password-placeholder": "Input password",
            "muted-text": "We'll never share your login with anyone else",
            "check-box": "Check out",
            "logIn-btn": "Login",
            "help-text":"Don`t have an account?",
            "link-text":"Go to register page"
        },
        handleSubmit: jest.fn(),
        getUsersDispatch: jest.fn(),
        changeStatusDispatch: jest.fn(),
    }

    beforeEach(() => {
        wrapper = mount(<Register {...props} />);
        instance = wrapper.instance();
    });

    it('should render correctly', () =>{
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('checkbox change calls handleCheckout', () =>{

        const spy = jest.spyOn(wrapper.instance(), "handleCheckout");
        wrapper.instance().forceUpdate();
        wrapper.find("#exampleCheck1").simulate("change",  spy);

        expect(spy).toBeCalled();
    });

    /*it('form submit calls handleSubmit', () =>{

        const spy = jest.spyOn(wrapper.instance(), "handleSubmit");
        wrapper.instance().forceUpdate();
        wrapper.find("form").simulate("submit",  spy);

        expect(spy).toBeCalled();
    });*/


});


