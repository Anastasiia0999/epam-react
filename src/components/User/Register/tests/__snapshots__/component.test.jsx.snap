// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should render correctly 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><form class=\\"col-4 my-5\\"><div class=\\"form-group-\\"><h4></h4><label for=\\"login-input\\">Login</label><input name=\\"login\\" type=\\"text\\" class=\\"form-control input-reg\\" id=\\"login-input\\" minlength=\\"8\\" aria-describedby=\\"loginHelp\\" placeholder=\\"Input login\\" value=\\"\\"><small id=\\"loginHelp\\" class=\\"form-text text-muted\\">We'll never share your login with anyone elseelse.</small></div><div class=\\"form-group\\"><label for=\\"input-password\\">Password</label><input name=\\"password\\" type=\\"password\\" class=\\"form-control input-reg\\" id=\\"input-password\\" minlength=\\"8\\" placeholder=\\"Input password\\" value=\\"\\"></div><div class=\\"form-check\\"><input type=\\"checkbox\\" class=\\"form-check-input\\" id=\\"exampleCheck1\\"><label class=\\"form-check-label\\" for=\\"exampleCheck1\\">Check out</label></div><button type=\\"submit\\" class=\\"btn btn-primary register my-3\\"></button></form></div></div>"`;
